cg/smart_contracts/ClearGDPR.sol

```diff
@@ -1,135 +1,152 @@
+        require(subjects[_subjectId].isErased == false);
+        _;
+    }
 
-  constructor() public {
-    controller = msg.sender; 
-  }
+    constructor() public {
+        controller = msg.sender;
+        processors = new address[](1);
+        processors[0] = controller;
+    }
 
-  function getRectifiedCount(bytes32 _subjectIdHash) public view returns(uint256){
-    return subjects[_subjectIdHash].rectificationCount;
-  }
+    function getRectificationCount(bytes32 _subjectId) public view returns(uint256){
+        return subjects[_subjectId].rectificationCount;
+    }
 
-  function getIsErased(bytes32 _subjectIdHash) public view returns(bool){
-    return subjects[_subjectIdHash].isErased;
-  }
+    function getIsErased(bytes32 _subjectId) public view returns(bool){
+        return subjects[_subjectId].isErased;
+    }
 
-  function getSubjectDataState(bytes32 _subjectIdHash, address _processorIdHash) public view returns(State){
-    return subjects[_subjectIdHash].statePerProcessor[_processorIdHash];
-  }
+    function getSubjectDataState(bytes32 _subjectId, address _processor) public view returns(State){
+        return subjects[_subjectId].statePerProcessor[_processor];
+    }
 
-  function getProcessors() public view returns(address[], uint256){
-    return (processors, processors.length);
-  }
+    function getProcessors() public view returns(address[], uint256){
+        return (processors, processors.length);
+    }
 
-  function setSubjectDataState(bytes32 _subjectIdHash, address _processorIdHash, State _state) public returns(bool){
-    subjects[_subjectIdHash].statePerProcessor[_processorIdHash] = _state;
-    return true;
-  }
+    function setSubjectDataState(bytes32 _subjectId, address _processor, State _state) public returns(bool){
+        subjects[_subjectId].statePerProcessor[_processor] = _state;
+        return true;
+    }
 
-  function setProcessors(address[] _newProcessors) public returns(bool){ //BUG IN HERE, CAN'T USE BYTES32
-    processors = new address[](_newProcessors.length);
-    processors = _newProcessors;
-    return true;
-  }
-  
-  function isProcessor(address _processor) public view returns(bool){ //0x1 for remix, since it's the only argument
-    for(uint256 i = 0; i < processors.length; i++){
-      if(processors[i] == _processor){
+    function setProcessors(address[] _newProcessors) public returns(bool){
+        processors = new address[](_newProcessors.length + 1);
+        processors[0] = controller;
+        for(uint256 i = 0; i < _newProcessors.length; i++){
+            require(_newProcessors[i] != controller);
+            for(uint256 j = i + 1; j < _newProcessors.length; j++){ // Check if the processors are unique
+                require(_newProcessors[i] != _newProcessors[j]);
+            }
+            processors[i + 1] = _newProcessors[i];
+        }
         return true;
-      }
+    } 
+    //processors = _newProcessors; //Interesting line
```

Comment:
Why is this line interesting? Shouldn't it just go away? ;)

Reply:
It can go away, but it's to remember us that we can assign array pointers in Solidity if the arrays have the same length. But I will remove it, and leave it in my Solidity notes